ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryPartition.java

```diff
@@ -40,12 +42,14 @@
   /**
    * Instantiate an AmbryPartition instance.
    * @param id the id associated with this partition.
+   * @param partitionClass the partition class that this partition belongs to
    * @param clusterManagerCallback the {@link ClusterManagerCallback} to use to make callbacks
    *                               to the {@link HelixClusterManager}
    * The initial state defaults to {@link PartitionState#READ_WRITE}.
    */
-  AmbryPartition(long id, ClusterManagerCallback clusterManagerCallback) {
+  AmbryPartition(long id, String partitionClass, ClusterManagerCallback clusterManagerCallback) {
```

Comment:
enum instead of string for partitionClass?

Reply:
Cannot be an enum because the partition classes can't be (and need not be) exhaustively listed in code